driver/src/price_finding/naive_solver.rs

```diff
@@ -164,7 +180,11 @@ impl PriceFinding for NaiveSolver {
 
             // apply fee to volumes account for rounding errors, moving them to
             // the fee token
-            for (i, order) in orders.iter().enumerate() {
+            let (order_1_index, order_2_index) =
+                matching_orders_indices.expect("fee price was nonzero so we should have a match");
+            for i in &[order_1_index, order_2_index] {
```

Comment:
Is the `&` necessary? I thought arrays implemented `IntoIter` as well but I may be wrong. If it does then you wouldn’t have to dereference `i`.

Reply:
I have to either add `&` or call `iter` manually. But either way `i` is a reference. Calling `into_iter` manually is a warning:
> warning: this method call currently resolves to `<&[T; N] as IntoIterator>::into_iter` (due to autoref coercions), but that might change in the future when `IntoIterator` impls for arrays are added.